cookbooks/bcpc/recipes/serial_console.rb

```diff
@@ -0,0 +1,59 @@
+def numeric?(data)
+        Float(data) != nil rescue false
+end
+
+serial_devices = {}
+# we are not interested in USB or any other serial at the moment
+serial_dev_type = "ttyS"
+serial_driver_info = "/proc/tty/driver/serial"
+File.open(serial_driver_info, "r") do |serials|
+    serials.each_line do |line|
+        if line.start_with? "serinfo:"
+            next
+        end
+        line_parts = line.split()
+        uart = line_parts[1]
+        if not uart.include? "uart:unknown"
+            port_assignment = line_parts[0]
+            port_number = port_assignment.chop
+            # get device baud
+            device_baud = `stty -F /dev/#{serial_dev_type}#{port_number} speed`
+            device_baud = device_baud.chomp
+            if not numeric?(device_baud)
+                next
+            end
+            device_name = "#{serial_dev_type}#{port_number}"
+            serial_devices[device_name] = device_baud
+        end
+    end
+end
```

Comment:
Could you wrap line 5 - 29 into a method in library util.rb? I think this recipe likes clean chef code.